src/test/java/com/desiato/puresynth/controllers/UserControllerTest.java

```diff
@@ -69,19 +73,34 @@ public void createUser_ShouldReturnCreatedUser() throws Exception {
                 .andExpect(jsonPath("$.email").value(uniqueEmail));
     }
 
+    @Transactional
     @Test
     public void deleteUser_ShouldDeleteUserAndReturnOk() throws Exception {
         // Given
-        String uniqueEmail = testAuthenticationHelper.generateUniqueEmail();
-        User createdUser = userService.createUser(uniqueEmail, "password123");
-        Long userId = createdUser.getId();
+        AuthenticatedUser authenticatedUser = testAuthenticationHelper.createAndAuthenticateUser();
+        AuthenticatedUser authenticatedUser2 = testAuthenticationHelper.createAndAuthenticateUser();
```

Comment:
The name is too vague. It should indicate it's an admin user, though you don't have this concept in code. Considering this, I think a validation in directly via repository (like you did in `UserServiceTest`) would be enough. Then you wouldn't need `UserServiceTest` at all.